packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -246,10 +257,17 @@ export class Pipeline extends PipelineBase {
       throw new Error('Only one of artifactBucket and crossRegionReplicationBuckets can be specified!');
     }
 
+    // Cross account keys must be set for key rotation to be enabled
+    if (props.enableKeyRotation && !props.crossAccountKeys) {
+      throw new Error('When setting \'enableKeyRotation\' you must also set \'crossAccountKeys\'');
+    }
```

Comment:
This condition is incorrect (`props. crossAccountKeys === undefined` means `true`, not `false`).

You should move this condition below, after you set the fields `crossAccountKeys` and `enableKeyRotation`.